learn/what_is_meilisearch/comparison_to_alternatives.md

```diff
@@ -75,31 +75,31 @@ Can't find a client you'd like us to support? [Submit your idea or vote for it](
 |   | Meilisearch | Algolia | Typesense | Elasticsearch |
 |---|:---:|:----:|:---:|:---:|
 | Typo tolerant  | ✅ | ✅ | ✅ | 🔶 <br>Needs to be specified by fuzzy queries |
-| Orderable ranking rules | ✅ | ✅ | 🔶 <br>Tie-breaking order is limited by a unique scoring rule | ❌|
-| Custom rules | ✅ | ✅ | 🔶 <br>Limited to one default sorting rule | 🔶 <br>Function score query
+| Orderable ranking rules | ✅ | ✅ | 🔶 <br>Field priority can be changed, but ranking rules order cannot be changed. | ❌|
+| Custom rules | ✅ | ✅ | ✅ | 🔶 <br>Function score query
 | Query field weights | ✅ | ✅ | ✅ | ✅ |
 | Synonyms | ✅ | ✅ | ✅ | ✅ |
 | Stop words | ✅ | ✅ | ❌ | ✅ |
 | Automatic language detection | ✅ | ✅ | ❌ | ❌ |
-| All language supports | ✅ | ✅ | ❌ <br> Only space separated | ✅ |
+| All language supports | ✅ | ✅ | ✅ | ✅ |
 
 ##### Security
 
 |   | Meilisearch | Algolia | Typesense | Elasticsearch |
 |---|:---:|:----:|:---:|:---:|
 | API Key Management | ✅ | ✅ | ✅ | ✅ |
-| Tenant tokens & multi-tenant indexes | ✅ <br> [Multitenancy support](/learn/security/tenant_tokens.md) | ✅  <br> Hard filters are not configurable per index for an end-user tenant key | ✅ <br> Hard filters are not configurable per index for an end-user tenant key | ✅ <br> Role-based |
```

Comment:
We addressed this in v0.23

Reply:
@gmourier I don't remember why we didn't put a simple checkmark there. How about changing it?

Reply:
I think it was when we released the tenant token feature; at the time v0.23 of Typesense was not released yet, and therefore, it was not possible to generate granular rules **for different indexes within a single JWT**.

We could rephrase it to make it clearer, Algolia does not have that capability.

From what I read from [typesense doc](https://typesense.org/docs/0.24.0/api/api-keys.html#generate-scoped-search-key), this is not possible to define different `filter_by` rules per `index A` and `indexB` on a Scoped Search Key. Am I right @jasonbosco? 

That's what we wanted to emphasize by saying, _"Hard filters are not configurable per index for an end-user tenant key"_

Thanks!

Reply:
@gmourier As of v0.23, you can embed a separate scoped API key inside each search inside a multi index search request, and have different filters for each index.

Reply:
I'm not even sure what a `Hard filter` is, so I can't help here.

Reply:
@gmourier 
Does typesense embedding a separate scoped API key inside each search inside a multi index search request answer your question on different filters for different indexes?

Reply:
Sorry to be late, @maryamsulemani97 and @jasonbosco

It seems it's not possible to define for one scoped API key a custom rule per accessed index, which makes me think that using Typesense, you are forced to forge `X` scoped API keys per user. `X` is the number of indexes accessible to that user that need a custom rule.

Let's say that I want to define rules for `user 1` when querying `index A` and `index B`

With Meilisearch, I can do it this way when forging a tenant token. You don't need to generate different tenant tokens for each index per user.

```jsonc
{
  "apiKeyUid": ...,
  "exp": ...,
  "searchRules": {
    "indexA": {
      "filter": "user_id = 1 AND is_premium = true"
    },
    "indexB": {
      "filter": "user_id = 1 AND published = true",
    }
  }
}
```

So I would prefer to keep that differentiator in the comparative table. Maybe it's more straightforward to say that competitors require `X` scoped api keys per user if the `X` accessible indexes need a unique rule.

@jasonbosco can you confirm that statement here?

Reply:
But that's just an implementation detail. "Tenant tokens" in Typesense are all generated client-side and so it's just one additional method call to generate an additional key for each index. You can then still send all of these keys in a single multi_search request. 

So the end result from a user's perspective is that they can apply "tenant scopes" at an index-level granularity both in Typesense and in Meilisearch. The exact syntax of how you achieve this end goal is what's different.

Reply:
So, can a "key generator" generate a key on the client side without calling any HTTP method of Typesense and restricting the documents that the key sees? And is it therefore possible for the "key generator" to do that by just using a Typensense key and derivate those filtered-key from it?

Reply:
> So, can a "key generator" generate a key on the client side without calling any HTTP method of Typesense and restricting the documents that the key sees? 

Yes. But you need to use an existing search API-key as the base, and use that to cryptographically generate the other restricted keys, and these restricted keys don't need HTTP calls to be created, since they are cryptographically generated using the base search API-key as the seed. Users need to generate these restricted API keys (or really any API keys) on the server-side and send it to the client side. 

> And is it, therefore, possible for the "key generator" to do that by just using a Typensense key and derivate those filtered-keys from it?

Yes. But you need to have access to the base search API key to generate additional filtered keys, like I mentioned above. And in a multi-tenant use-case, users don't expose the base search API key to the frontend, instead expose the restricted API keys. 

It feels like we're going off on a tangent here about how these keys work in Typesense (again an implementation detail). To re-iterate the key point we were discussing in this thread previously: Typesense has the ability for users to apply "tenant scopes" at an index-level granularity and so the statement that is currently public is wrong / outdated.

Reply:
Yup! Let's merge this PR as it is now.